src/Meilisearch/TaskEndpoint.cs

```diff
@@ -0,0 +1,107 @@
+
+        /// <summary>
+        /// Gets one task.
+        /// </summary>
+        /// <param name="taskUid">Uid of the index.</param>
+        /// <param name="cancellationToken">The cancellation token for this call.</param>
+        /// <returns>Returns the task.</returns>
+        public async Task<TaskInfo> GetTaskAsync(int taskUid, CancellationToken cancellationToken = default)
+        {
+            return await this.http.GetFromJsonAsync<TaskInfo>($"/tasks/{taskUid}", cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets the tasks from an index.
+        /// </summary>
+        /// <param name="indexUid">Uid of the index.</param>
+        /// <param name="cancellationToken">The cancellation token for this call.</param>
+        /// <returns>Returns a list of tasks of an index.</returns>
+        public async Task<Result<IEnumerable<TaskInfo>>> GetIndexTasksAsync(string indexUid, CancellationToken cancellationToken = default)
+        {
+            return await this.http.GetFromJsonAsync<Result<IEnumerable<TaskInfo>>>($"/indexes/{indexUid}/tasks", cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Get one task from an index.
+        /// </summary>
+        /// <param name="indexUid">Uid of the index.</param>
+        /// <param name="taskUid">Uid of the task.</param>
+        /// <param name="cancellationToken">The cancellation token for this call.</param>
+        /// <returns>Return the task of the index.</returns>
+        public async Task<TaskInfo> GetIndexTaskAsync(string indexUid, int taskUid, CancellationToken cancellationToken = default)
+        {
+            return await this.http.GetFromJsonAsync<TaskInfo>($"/indexes/{indexUid}/tasks/{taskUid}", cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Waits until the asynchronous task was done.
+        /// </summary>
+        /// <param name="taskUid">Unique identifier of the asynchronous task.</param>
+        /// <param name="timeoutMs">Timeout in millisecond.</param>
+        /// <param name="intervalMs">Interval in millisecond between each check.</param>
+        /// <param name="cancellationToken">The cancellation token for this call.</param>
+        /// <returns>Returns the task info of finished task.</returns>
+        public async Task<TaskInfo> WaitForTaskAsync(
+            int taskUid,
+            double timeoutMs = 5000.0,
+            int intervalMs = 50,
+            CancellationToken cancellationToken = default)
+        {
+            DateTime endingTime = DateTime.Now.AddMilliseconds(timeoutMs);
+
+            while (DateTime.Now < endingTime)
+            {
+                var task = await this.GetTaskAsync(taskUid, cancellationToken).ConfigureAwait(false);
+
+                if (task.Status != "enqueued" && task.Status != "processing")
```

Comment:
Am I too annoying about this idea of adding these as a constant? I've bothered @alallema with that question in the Java PR as well...

Reply:
I'm not sure to understand how to make it `const` since the value changes in the loop

Reply:
Oh ok, my bad I misunderstood, I thought ou wanted to make `task` constant

Reply:
Yeah, @juchom what do you think about it?

Reply:
You could go with something like `ClaimTypes` in framework

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Security.Claims/src/System/Security/Claims/ClaimTypes.cs

Reply:
Thanks for the clue! Would be better to do it in another PR :)